test/parallel/test-http-parser-lazy-loaded.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+// Monkey patch before requiring anything
+class DummyParser {
+  constructor(type) {
+    this.test_type = type;
+  }
+}
+DummyParser.REQUEST = Symbol();
+process.binding('http_parser').HTTPParser = DummyParser;
+
+const common = require('../common');
+const assert = require('assert');
+const { fork } = require('child_process');
+const { parsers } = require('_http_common');
+
+function doTest() {
+  // Test _http_common was not loaded before monkey patching
+  const parser = parsers.alloc();
+  assert.strictEqual(parser instanceof DummyParser, true);
+  assert.strictEqual(parser.test_type, DummyParser.REQUEST);
+}
+
+if (process.argv[2] === 'child') {
+  doTest();
+  return;
+}
+
+// Test in our process
+doTest();
+// Test in a child process with IPC (specific case of https://github.com/nodejs/node/issues/23716)
```

Comment:
You could execute this block of code only if `process.argv[2] !== 'child'`, then you could get rid of the logic above, only calling `doTest()` once, unconditionally. Then, since it's only called once, you could get rid of `doTest()`, and inline its body 😄

Reply:
Yeah, my first pass had the child process sending a message back to say things succeeded, so it was more complicated, and after I submitted my last change I did notice this could be simplified.  Will do, since I'm making the other changes =).